src/com/owncloud/android/ui/dialog/RateMeDialog.java

```diff
@@ -0,0 +1,165 @@
+     */
+    public static RateMeDialog newInstance(String packageName, boolean cancelable) {
+        RateMeDialog fragment = new RateMeDialog();
+        Bundle args = new Bundle();
+        args.putBoolean(ARG_CANCELABLE, cancelable);
+        args.putString(APP_PACKAGE_NAME, packageName);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Create a view by inflating desired layout
+        View v = inflater.inflate(R.layout.rate_me_dialog, container,  false);
+
+        Button rateNowButton = (Button) v.findViewById(R.id.button_rate_now);
+        Button laterButton = (Button) v.findViewById(R.id.button_later);
+        Button noThanksButton = (Button) v.findViewById(R.id.button_no_thanks);
+
+        rateNowButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String packageName = getArguments().getString(APP_PACKAGE_NAME);
+                Uri uri = Uri.parse("market://details?id=" + packageName);
+                Intent goToMarket = new Intent(Intent.ACTION_VIEW, uri);
+
+                /// To count with Play market back stack, After pressing back button,
+                /// to taken back to our application, we need to add following flags to intent.
+                int flags = Intent.FLAG_ACTIVITY_NO_HISTORY |
+                        Intent.FLAG_ACTIVITY_MULTIPLE_TASK;
+                if (Build.VERSION.SDK_INT >= 21)
+                {
+                    flags |= Intent.FLAG_ACTIVITY_NEW_DOCUMENT;
+                }
+                else
+                {
+                    flags |= Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET;
+                }
+                goToMarket.addFlags(flags);
+
+                try {
+                    startActivity(goToMarket);
+                } catch (ActivityNotFoundException e) {
+                    getActivity().startActivity(new Intent(Intent.ACTION_VIEW,
+                            Uri.parse("http://play.google.com/store/apps/details?id=" + packageName)));
+                }
+                dialog.dismiss();
+            }
+        });
+
+        laterButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                SharedPreferences preferences = getActivity().getSharedPreferences("app_rater", 0);
+                SharedPreferences.Editor editor = preferences.edit();
+                editor.putLong("date_neutral", System.currentTimeMillis());
+                editor.apply();
+                dialog.dismiss();
+            }
+        });
+
+        noThanksButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                SharedPreferences preferences = getActivity().getSharedPreferences("app_rater", 0);
```

Comment:
From this class you could access to the preferences names constants declared in AppRater